tests/cases/events_test.py

```diff
@@ -136,3 +138,25 @@ def callback(event):
             self.assertEqual(events.daemon.eventQueue.qsize(), 0)
             self.assertEqual(self.ctr, 3)
             self.assertEqual(self.responses, ['foo'])
+            events.daemon.stop()
+
+    @mock.patch.object(events, 'daemon', new=events.ForegroundEventsDaemon())
+    def testForegroundDaemon(self):
```

Comment:
This works for the context manager, thanks.

One last worry is that this doesn't actually test that we are using the ForegroundEventsDaemon for events.daemon. Am I right to worry about that?  Couldn't this test code also work if the standard async daemon is used?

I guess it comes down to trusting that the mock.patch works.  Or, trust, but verify.  Perhaps just a type check of events.daemon, that it is a ForegroundEventsDaemon?